pkgs/applications/virtualization/libnvidia-container/makefile.patch

```diff
@@ -0,0 +1,75 @@
+-LIB_CPPFLAGS       = -DNV_LINUX -isystem $(DEPS_DIR)$(includedir) -include $(BUILD_DEFS)
++LIB_CPPFLAGS       = -DNV_LINUX -isystem $(includedir) -include $(BUILD_DEFS) -I $(SRCS_DIR)
+ LIB_CFLAGS         = -fPIC
+-LIB_LDFLAGS        = -L$(DEPS_DIR)$(libdir) -shared -Wl,-soname=$(LIB_SONAME)
++LIB_LDFLAGS        = -L$(libdir) -shared -Wl,-soname=$(LIB_SONAME)
+ LIB_LDLIBS_STATIC  = -l:libnvidia-modprobe-utils.a
+ LIB_LDLIBS_SHARED  = -ldl -lcap
+ ifeq ($(WITH_NVCGO), yes)
+@@ -165,7 +165,7 @@ else
+ LIB_LDLIBS_STATIC  += -l:libelf.a
+ endif
+ ifeq ($(WITH_TIRPC), yes)
+-LIB_CPPFLAGS       += -isystem $(DEPS_DIR)$(includedir)/tirpc -DWITH_TIRPC
++LIB_CPPFLAGS       += -isystem $(includedir)/tirpc -DWITH_TIRPC
+ LIB_LDLIBS_STATIC  += -l:libtirpc.a
+ LIB_LDLIBS_SHARED  += -lpthread
+ endif
+@@ -231,7 +231,7 @@ $(LIB_SHARED): $(LIB_OBJS)
+ 
+ $(LIB_STATIC_OBJ): $(LIB_OBJS)
+ 	# FIXME Handle user-defined LDFLAGS and LDLIBS
+-	$(LD) -d -r --exclude-libs ALL -L$(DEPS_DIR)$(libdir) $(OUTPUT_OPTION) $^ $(LIB_LDLIBS_STATIC)
++	$(LD) -d -r --exclude-libs ALL -L$(libdir) $(OUTPUT_OPTION) $^ $(LIB_LDLIBS_STATIC)
+ 	$(OBJCPY) --localize-hidden $@
+ 	$(STRIP) --strip-unneeded -R .comment $@
+ 
+@@ -242,7 +242,7 @@ $(BIN_NAME): $(BIN_OBJS)
+ ##### Public rules #####
+ 
+ all: CPPFLAGS += -DNDEBUG
+-all: shared static tools
++all: shared tools
+ 
+ # Run with ASAN_OPTIONS="protect_shadow_gap=0" to avoid CUDA OOM errors
+ debug: CFLAGS += -pedantic -fsanitize=undefined -fno-omit-frame-pointer -fno-common -fsanitize=address
+@@ -258,15 +258,15 @@ static: $(LIB_STATIC)($(LIB_STATIC_OBJ))
+ 
+ deps: $(LIB_RPC_SRCS) $(BUILD_DEFS)
+ 	$(MKDIR) -p $(DEPS_DIR)
+-	$(MAKE) -f $(MAKE_DIR)/nvidia-modprobe.mk DESTDIR=$(DEPS_DIR) install
++	$(MAKE) -f $(MAKE_DIR)/nvidia-modprobe.mk DESTDIR="" install
+ ifeq ($(WITH_NVCGO), yes)
+-	$(MAKE) -f $(MAKE_DIR)/nvcgo.mk DESTDIR=$(DEPS_DIR) MAJOR=$(MAJOR) VERSION=$(VERSION) LIB_NAME=$(LIBGO_NAME) install
++	$(MAKE) -f $(MAKE_DIR)/nvcgo.mk DESTDIR="" MAJOR=$(MAJOR) VERSION=$(VERSION) LIB_NAME=$(LIBGO_NAME) install
+ endif
+ ifeq ($(WITH_LIBELF), no)
+-	$(MAKE) -f $(MAKE_DIR)/elftoolchain.mk DESTDIR=$(DEPS_DIR) install
++	$(MAKE) -f $(MAKE_DIR)/elftoolchain.mk DESTDIR="" install
+ endif
+ ifeq ($(WITH_TIRPC), yes)
+-	$(MAKE) -f $(MAKE_DIR)/libtirpc.mk DESTDIR=$(DEPS_DIR) install
++	$(MAKE) -f $(MAKE_DIR)/libtirpc.mk DESTDIR="" install
+ endif
+ 
+ install: all
+@@ -274,11 +274,9 @@ install: all
+ 	# Install header files
+ 	$(INSTALL) -m 644 $(LIB_INCS) $(DESTDIR)$(includedir)
+ 	# Install library files
+-	$(INSTALL) -m 644 $(LIB_STATIC) $(DESTDIR)$(libdir)
+ 	$(INSTALL) -m 755 $(LIB_SHARED) $(DESTDIR)$(libdir)
+ 	$(LN) -sf $(LIB_SONAME) $(DESTDIR)$(libdir)/$(LIB_SYMLINK)
+ ifeq ($(WITH_NVCGO), yes)
+-	$(INSTALL) -m 755 $(DEPS_DIR)$(libdir)/$(LIBGO_SHARED) $(DESTDIR)$(libdir)
+ 	$(LN) -sf $(LIBGO_SONAME) $(DESTDIR)$(libdir)/$(LIBGO_SYMLINK)
+ endif
+ 	$(LDCONFIG) -n $(DESTDIR)$(libdir)
```

Comment:
Could this patch be replaced with a call to `substituteAllInPlace` in `postPatch`?

Reply:
I don't think so, but maybe I'm not understand what that would buy us here. Would the approach be to replace `$(DEPS_DIR)` with `@depsDir@` and then define `depsDir = "";` and then call `substituteAllInPlace`?

Reply:
Yeah, so instead of maintaining a patch that needs rebasing every release we can just use `substituteAll`, or even just `sed`

Reply:
Alright, ended up using `sed` here. Also went ahead and templated the modprobe version in `modprobe.patch` using `substituteAll`.